cmd/internal/cli/singularity.go

```diff
@@ -50,6 +50,49 @@ const (
 	envPrefix = "SINGULARITY_"
 )
 
+// checkForInvalidShortFlags ensures that supported short flags are used with
+// the singularity command. Cobra natively use the pflag type instead of the
+// flag type and as a result has limited support for short flags (i.e., flags
+// with a single dash such as "-foo"). As a result, Cobra does not check if
+// short flags are valid or not. This function addresses this gap by manually
+// checking the validity of the short flags that are passed in.
+// The expected format of the command is:
+// singularity <singularitY_short_flags> <singularity_long_flags> singularity_cmd
+func checkForInvalidSingularityShortFlags() {
+	// Unfortunately, it seems that Cobra does not maintain a structure to
```

Comment:
flags are kept in the pflag.Flag struct. Each command has a Flags() method that returns a pflag.FlatSet, which is the aggregation of flags. The pflag.Flag struct contains a field called "Shorthand" which is the abbreviated version of the flag. For example "debug" says that Name is "debug" and Shorthand is "d". Is that not the case?

pflag supports mixing short flags, but there are some rules as to how that happens.

What problem is this trying to solve?

Reply:
I was trying the solve the issue mentioned in the PR. The issue is only related to the short flags for the singularity command (as explained in the comments) so i do not believe that the code in place for the pflag.Flag structure is actually were the work needs to happen. But it does not matter, Cedric will address that problem in the context of #3230